src/sentry/tasks/reports.py

```diff
@@ -638,46 +638,118 @@ def deliver_organization_user_report(timestamp, duration, organization_id, user_
         duration,
         organization,
         user,
-        reduce(
-            merge_reports,
-            reports,
-        )
+        reports,
     )
 
     if features.has('organizations:reports:deliver', organization):
         message.send()
 
 
-IssueList = namedtuple('IssueList', 'count issues')
-IssueStatistics = namedtuple('IssueStatistics', 'events users')
+Point = namedtuple('Point', 'resolved unresolved')
+DistributionType = namedtuple('DistributionType', 'label color')
 
 
-def rewrite_issue_list((count, issues), fetch_groups=None):
-    # XXX: This only exists for removing data dependency in tests.
-    if fetch_groups is None:
-        fetch_groups = Group.objects.in_bulk
+def series_map(function, series):
+    return [(timestamp, function(value)) for timestamp, value in series]
 
-    instances = fetch_groups([id for id, _ in issues])
 
-    def rewrite((id, statistics)):
-        instance = instances.get(id)
-        if instance is None:
-            logger.debug("Could not retrieve group with key %r, skipping...", id)
-            return None
-        return (instance, IssueStatistics(*statistics))
+colors = [
+    '#696dc3',
+    '#6288ba',
+    '#59aca4',
+    '#99d59a',
+    '#daeca9',
+]
 
-    return IssueList(
-        count,
-        filter(None, map(rewrite, issues)),
-    )
 
+def build_project_breakdown_series(reports):
+    Key = namedtuple('Key', 'label url color data')
 
-Point = namedtuple('Point', 'resolved unresolved')
-DistributionType = namedtuple('DistributionType', 'label color')
+    def get_legend_data(report):
+        series, _, _, _, (filtered, rate_limited) = report
+        return {
+            'events': sum(sum(value) for timestamp, value in series),
+            'filtered': filtered,
+            'rate_limited': rate_limited,
+        }
 
+    # Find the reports with the most total events. (The number of reports to
+    # keep is the same as the number of colors available to use in the legend.)
+    instances = map(
+        operator.itemgetter(0),
+        sorted(
+            reports.items(),
+            key=lambda (instance, report): sum(sum(values) for timestamp, values in report[0]),
+            reverse=True,
+        ),
+    )[:len(colors)]
+
+    # Starting building the list of items to include in the report chart. This
+    # is a list of [Key, Report] pairs, in *ascending* order of the total sum
+    # of values in the series. (This is so when we render the series, the
+    # largest color blocks are at the bottom and it feels appropriately
+    # weighted.)
+    selections = map(
+        lambda (instance, color): (
+            Key(
+                u'{} / {}'.format(
+                    instance.team.name,
```

Comment:
I think we should just use ``instance.slug`` here (continue to deprecate team names).